### ISSUE TITLE
How to augment data to increase the number of instances being trained on?

### ISSUE THREAD

**Issue by mohamedamrali1993** [facebookresearch/detectron2#3701]
### Discussed in https://github.com/facebookresearch/detectron2/discussions/3698
<sup>Originally posted by **mohamedamrali1993** November 15, 2021</sup>
<div type='discussions-op-text'>
## Instructions To Reproduce the Issue:
Linux 18.04
Cuda 11.3
Pytorch
Detectron2

Hi, I am having a hard time understanding "Augmentation" in the context of Detectron2. It seems like the [Data Augmentation page](https://detectron2.readthedocs.io/en/latest/tutorials/augmentation.html) is striving to create variations within the dataset, but not necessarily "augmenting" as in increasing the size of the dataset. i.e. if I have a dataset with 100 images and I apply RandomFlip() transformation to 5% of the data, I still end up with 100 instances to train on. Not 105! Am I understanding this correctly or is it supposed to increase the instances, and I am just doing something wrong?

## Expected behavior:
I tracked down the process of data augmentation, and at no point did I see data being appended to a larger query (with original images + transformed images), which is eventually sent to the model. Here is what I understand so far:

1) The trainer calls on to the **build_train_loader()**, which returns the **build_detection_test_loader()** with the "mapper" as an argument.
2) We can custom make our own "mapper" and include the appropriate transformations we want.
3) The mapper needs to transform images sequentially, and consequently transform annotations (such as bbox, masks, and keypoints) using the **utils.transform_instance_annotations(annotation, transform, size)**
4) Finally **utils.annotations_to_instances(annotations, size, mask_format)** creates the instances that are forwarded to the model.

I would like to increase the number of images that I am training on, and their respective transformed labels. Is this doable using Detectron2 and is it a good idea for Mask R-CNN? FAIR's 2018 Mask R-CNN paper has a small section in Appendix B that talks about "Train-time augmentation" where they scale images (if I am understanding things correctly), but I am not sure if they are actually training on the original images or "original images + scaled images." I am expecting the latter!

If this can be done using Detectron2 then please advise (including code examples), if not, I would imagine this would be a good enhancement!?

I am not the first one to report/ask about this, someone else asked about it in issue [#1763](https://github.com/facebookresearch/detectron2/issues/1763#issuecomment-659258920)
</div>
